Add Order Summary card to Checkout sidebar

Aljullu opened the pull request:
Closes #1486.
Closes #1304.

### Screenshots
![imatge](https://user-images.githubusercontent.com/3616980/76559169-ed799500-649e-11ea-8b5f-b4e6c8860fa0.png)

### How to test the changes in this Pull Request:

1. Create a page with the _Checkout_ block and verify it has the _Order Summary_ in the sidebar.
2. Verify there are no visual regressions in the _Cart_ block.

Aljullu commented on assets/js/base/components/cart-checkout/index.js:
```diff
@@ -0,0 +1,7 @@
+export { default as ProductName } from './product-name';
+export { default as ProductPrice } from './product-price';
+export { default as ProductSaleBadge } from './product-sale-badge';
+export { default as ProductVariationData } from './product-variation-data';
```
I noticed some of these blocks are very similar to the ones in `atomic/components/product/`. I think it might be worth checking if they can be merged, but I left it out of this PR because it was already quite big and I don't think that's something blocking.

haszari replied:
Agree, this is good follow up for later. Great that cart & checkout are sharing these components now though :)

haszari commented on assets/js/base/components/cart-checkout/product-sale-badge/index.js:
```diff
@@ -0,0 +1,35 @@
+import './style.scss';
+
+/**
+ * Returns a low stock badge for a line item.
```
Comment pasteo :)

Aljullu replied:
Removed in 4a60a67b57a8d8d5f40fc7467b79c69c973fd942.

haszari commented on assets/js/blocks/cart-checkout/cart/full-cart/style.scss:
```diff
@@ -321,7 +273,7 @@ table.wc-block-cart-items {
 				}
 
-				.wc-block-cart-item__sale-badge {
+				.wc-block-sale-badge {
```
Should some of these styles be in the components directly? Probably not a big deal, there's a little mixing of concerns here - the cart is overriding/tweaking component styles.

Aljullu replied:
I don't think so, I left them here in purpose. Even though they refer to the `ProductSaleBadge` component, they are only relevant under the _Cart_ block, so I think they belong to the _Cart_ CSS.

haszari commented on assets/js/blocks/cart-checkout/checkout/order-summary-item.js:
```diff
@@ -0,0 +1,86 @@
+			<div className="wc-block-order-summary-item__total-price">
+				<ProductPrice
+					currency={ currency }
+					value={ purchasePrice * quantity }
```
Does this work correctly with the different tax options?

Aljullu replied:
I think so, those prices honor the tax settings.

I found an issue, though: #1963. But since it's affecting the _Cart_ too, I don't think it should block this PR.

haszari commented on assets/js/blocks/cart-checkout/checkout/order-summary-item.js:
```diff
@@ -0,0 +1,86 @@
+		variation,
+	} = cartItem;
+
+	const { quantity } = useStoreCartItemQuantity( cartItem );
```
I don't think you need this hook - you can just accept a static quantity prop from parent component (which should have all the item details).

`useStoreCartItemQuantity` hook is for changing quantity or removing item :)

Aljullu replied:
Oh, good catch. I missed that `cartItem` has the `quantity` attribute. Fixed in 87214e31c0bb18ccdcd9e22ab76ab278f95e48eb.

haszari commented:
@Aljullu One more thing .. is it possible to give the description / summary text more room? They all wrap and it looks like this is unnecessary for the data I have. (e.g. `This is a simple product`)

<img width="264" alt="Screen Shot 2020-03-13 at 12 41 52 PM" src="https://user-images.githubusercontent.com/4167300/76577324-cd42e800-6529-11ea-99c2-b94b59ca4621.png">

garymurray commented:
> The circled quantity numbers seem a little overkill when there's 1x of a product, I'm guessing this is the intended design. Could consider showing number only when there are multiple of the item cc @garymurray. (Note this is exacerbated by my red customisation!!)

Let's leave that as is for now. Any changes here can be made later in future iterations.

Aljullu commented:
> Focus highlight on order summary "collapse" heading element has different padding/margin on right/left.

Fixed in 7d4fd6eb2eb9458ef01b38cdfcd8362115b0fa22.

> Hard-coded text color - it would be great if we can avoid this and/or use theme colors. In my cart I'm seeing a mix of black-on-white (opinionated blocks) vs. red (storefront customiser).

That's a tricky issue that I would like to handle in another PR, the hardcoded colors are there because in the designs they differ from the default color. 

I also noticed the _Cart_ block completely ignores the _Text color_ preference, while the _Checkout_ block applies it inconsistently. I created an issue to discuss what's the best solution for this: #1964.

> Hard error when a bad coupon is entered - I think this is already logged as an issue.

Right, I think that's not related to this PR.

> One more thing .. is it possible to give the description / summary text more room? 

Good idea. I refactored the CSS in c575d54564b920b26866195a40bd7937aea27e7e.
